pennylane/ops/op_math/composite.py

```diff
@@ -160,6 +160,13 @@ def eigendecomposition(self):
 
         return self._eigs[self.hash]
 
+    @property
+    def has_diagonalizing_gates(self):
+        if self.has_overlapping_wires:
+            return True
+
+        return all(op.has_diagonalizing_gates for op in self)
+
```

Comment:
```suggestion
        return self.has_overlapping_wires or self.has_matrix
```

For `Sum` and `Prod` we use the eigendecomposition of the matrix.

Reply:
I only found the code in `composite.py` that uses the eigendecomposition if `has_overlapping_wires=True` and the diagonalizing gates of the operands otherwise :thinking:

Reply:
Oops.  My suggestion was indeed wrong.  I made a corrected suggestion above.